megamek/src/megamek/client/ui/swing/boardview/BoardView1.java

```diff
@@ -6661,16 +6663,20 @@ public void changeTheme() {
 
         if (selectedTheme == null) {
             return;
-
-        } else if (selectedTheme.equals("(Original Theme)")) {
-            for (Coords c: allBoardHexes()) {
+        }
+        final List<Coords> allCoords = allBoardHexes();
+        if (allCoords == null) {
+            return;
+        }
+        if (selectedTheme.equals("(Original Theme)")) {
+            for (Coords c: allCoords) {
                 IHex hex = board.getHex(c);
                 hex.resetTheme();
                 board.setHex(c, hex);
             }
 
         } else {
-            for (Coords c: allBoardHexes()) {
+            for (Coords c: allCoords) {
```

Comment:
Why make the change here instead of just fixing the allBoardHexes() erroneous null return and replacing it with Collections.emptyList() (the return value of ArrayList can be swapped to List without issue)

Reply:
Unknown whether allBoardHexes() is used elsewhere and what effects that may have (e.g. other places may depend on it being null).

Reply:
Yes, that would be better.

Reply:
It's used in one other place which also fails to check for a null return value.

Reply:
More bugs fixed = better.